OL02 - 앱 회원 가입 기능 구현 (feature/OL02-signup_login)

ch-Yoon opened the pull request:
## 개요
- 앱 회원 가입 기능 구현(자체 서버와의 통신)
- 앱 로그인 기능 구현(이미 가입된 회원 시 자동 로그인)
## 작업내용
- 현재 서버가 완성되지 않았기에, shared preference를 통해 가상 flow만 구성
- 세션 존재O -> 카카오 인증 성공 -> 회원가입 여부 확인 -> 가입되었을 시 업주 상세 화면으로 이동
- 세션 존재O -> 카카오 인증 성공 -> 회원가입 여부 확인 -> 가입이 안되었을 시 로그인 화면에서 대기
- 세션 존재O -> 카카오 인증 실패 -> 에러 토스트 메시지 출력
- 세션 존재X  -> 로그인 화면에서 대기
## 특이사항
- 현재 서버가 완성되지 않아, shared preference를 통해 가상 flow만 확인
- 추후 Retrofit 활용하여 서버와 통신 후 반영 필요함

resolved #17

ch-Yoon committed:
OL02 - create basic structure of sign up screen

- Created the OwnerSignUpActivity
- Created the OwnerSignupContract
- Created the OwnerSignUpPresenter

ch-Yoon committed:
OL02 - add widget of sign up layout

ch-Yoon committed:
OL02 - implements sign up flow

- used the server of virtual
- used the user id and access key
- implemented the virtual sign up flow with the user id, accesskey

ch-Yoon committed:
OL02 - complete sign up virture logic

ch-Yoon committed:
update the wrong parts

- changed the hard code to resource
- when the presenter communicated with the view, check the attach view

lunar999 commented on app/src/main/java/com/aone/menurandomchoice/GlobalApplication.java:
```diff
@@ -2,7 +2,7 @@
 import android.app.Application;
 
-import com.aone.menurandomchoice.data.oauth.KakaoSDKAdapter;
+import com.aone.menurandomchoice.repository.oauth.KakaoSDKAdapter;
```
good 👍

lunar999 commented on app/src/main/java/com/aone/menurandomchoice/repository/server/ServerDataRepository.java:
```diff
@@ -0,0 +1,64 @@
+    @Override
+    public void requestSignedUpCheck(long userId, @NonNull OnSignedUpCheckListener onSignedUpCheckListener) {
+        try {
+            Thread.sleep(500);
```
왜 슬립을 할까요? 이유가 있나요?

ch-Yoon replied:
현재 가상의 시나리오로 구성된 코드인데요, 서버와의 통신이 이루어진다면 어느정도의 딜레이가 발생될거라고 생각했습니다. 그래서 실제처럼 약간의 딜레이 및 화면이 넘어가는 작업을 실제로 보기 위해 sleep을 주었습니다.

실제 서버와 통신할때는 저 코드는 필요가 없습니다!
서버와 통신하는 코드를 작성할 때 해당 코드들을 전부 삭제하고 꼼꼼히 처리하겠습니다.

lunar999 commented on app/src/main/java/com/aone/menurandomchoice/repository/server/ServerDataRepository.java:
```diff
@@ -0,0 +1,64 @@
+                onSignedUpCheckListener.onNotSignUp();
+            }
+        } catch (Exception e){
+            e.printStackTrace();
```
Logcat으로 변환필요

ch-Yoon replied:
이 부분은 지금처럼 가상의 테스트코드가 아닌, 서버와의 통신을 처리할 때 exception처리를 꼼꼼하게 해서 반영하겠습니다!

lunar999 commented on app/src/main/java/com/aone/menurandomchoice/views/ownerlogin/OwnerLoginPresenter.java:
```diff
@@ -32,17 +37,74 @@ public boolean isNeedKakaoSDKLoginScreen(int requestCode, int resultCode, @Nulla
+                break;
+        }
+    }
+
```
이 함수를 enum을 이용하면 이렇게 하지 않아도 해결딜꺼 같긴한데요
SERVER_ERROR(-1, R.string.activity_owner_toast_server_error)  이런 형식으로 만들면 이 switch 문이 더 간단해질꺼 같은데 어떠세요?

ch-Yoon replied:
이부분에서 고민을 참 많이했습니다. 

1. enum에서 string.xml의 id값을 갖고서 error처리를 하는 방법
2. 현재의 코드처럼 사용하는 방법
3. 별도의 error 객체를 create하는 factory class를 만드는 방법

어떤 기준으로 선택하는 것이 좋을까요? 현업에서는 comment해주신 방법으로 많이 사용하나요?

ch-Yoon replied:
말씀해주신대로 error 메시지를 핸들링하여 뷰에 송신하는 코드를
enum을 활용하여 처리했습니다.

b8b7c4652e854f27243f7eb9465a800b282daeac

ch-Yoon committed:
OL02 - update error handling method in presenter

- Remove multiple conditional statements
- Using the resource id in KakaoLoginError

ch-Yoon committed:
OL02 - Changed the e.printStack to listener.onSignUpFail()

ch-Yoon committed:
OL02 - delete Thread.sleep code
